olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp

```diff
@@ -279,6 +282,29 @@ bool PartitionsCacheRepository::ClearPartitionMetadata(
   return cache_->RemoveKeysWithPrefix(key);
 }
 
+bool PartitionsCacheRepository::IsPartitionCached(
+    const boost::optional<int64_t>& catalog_version,
+    const std::string& partition_id, const std::string& layer_id) {
+  std::string hrn(hrn_.ToCatalogHRNString());
+  auto key = CreateKey(hrn, layer_id, partition_id, catalog_version);
+  OLP_SDK_LOG_DEBUG_F(kLogTag, "IsPartitionCached -> '%s'", key.c_str());
+  if (cache_->Contains(key)) {
+    auto cached_partition =
+        cache_->Get(key, [](const std::string& serialized_object) {
+          return parser::parse<model::Partition>(serialized_object);
+        });
+
+    if (cached_partition.empty()) {
+      return false;
+    }
+    auto partition = boost::any_cast<model::Partition>(cached_partition);
+    auto data_key = CreateKey(hrn, layer_id, partition.GetDataHandle());
+    OLP_SDK_LOG_DEBUG_F(kLogTag, "IsPartitionCached data -> '%s'", key.c_str());
+    return cache_->Contains(data_key);
+  }
```

Comment:
Looks misleading to me. This method should check is metadata cached only.
Also, I think, calling `cache_->Get()` after `cache_->Contains()` is an overhead.

Reply:
We need to check if the data handle is also cached. Knowing that the metadata is cached does not bring anything.

Reply:
Yeah, but `PartitionsCacheRepository` isn't a best place to do it.
I would prefer to check data handle in `VersionedLayerClientImpl` instead.

Reply:
moved